alternator/rjson.cc

```diff
@@ -19,6 +19,30 @@
  * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <seastar/core/thread.hh>
+
+static inline void maybe_yield() {
+    if (seastar::thread::running_in_thread() && seastar::thread::should_yield()) {
+        seastar::thread::yield();
+    }
+}
```

Comment:
Soon it can be replaced with `co_await make_ready_future<>();`

Reply:
running_in_thread() can be misleading.

return async([] {
    foo().then([] {
            // check running_in_thread()
    });
 });

can return different values depending on choices made by future::then. So better not to rely on it.
}